docs/adr/0004-jest-27-support.md

```diff
@@ -0,0 +1,55 @@
+# 4. Jest 27 Support
+
+Date: 2021-07-30
+
+## Status
+
+accepted
+
+## Context
+
+The community has been asking for the ability to use a newer version on the Jest library (v27.0.0). Stencil does runtime checks on the version of the Jest library that the user is currently using, and fails up front if the user does not have what it deems to be an acceptable version installed.
+
+Jest 27 introduces [multiple breaking changes](https://jestjs.io/blog/2021/05/25/jest-27), including:
+
+- Moving away from [core Jasmine testing infrastructure](https://jestjs.io/blog/2021/05/25/jest-27#flipping-defaults) to `jest-circus`)
+- Changing the default runtime environment from JSDom to Node
+- Changing the signature of `getCacheKey` for file transformations
+- Inclusion of it's own type declarations, deprecating the community supported `@types/jest` package
```

Comment:
```suggestion
- Inclusion of its own type declarations, deprecating the community supported `@types/jest` package
```

I have no one to blame but myself for this 😆 

(since I wrote this one)